hash_reader.go

```diff
@@ -133,43 +172,98 @@ func (hashReader *hashReader) TimeValue() (time.Time, error) {
 }
 
 func (hashReader *hashReader) StringValue() (string, error) {
-	panic("implement me")
 	return hashReader.ionReader.StringValue()
 }
 
 func (hashReader *hashReader) ByteValue() ([]byte, error) {
-	panic("implement me")
 	return hashReader.ionReader.ByteValue()
 }
 
 func (hashReader *hashReader) Sum(b []byte) []byte {
-	panic("implement me")
+	hashReader.hasher.digest()
+}
+
+func (hashReader *hashReader) traverse() error {
+	for hashReader.Next() {
```

Comment:
Is this correct?

Reply:
In dot net, this.MoveNext() returns an ion type while hashReader.Next() returns a bool in go.

I see that textreader's implementation of Next() returns false when the token reading is done so it seems reasonable to expect the hashReader.Next() loop to keep going when it returns true.